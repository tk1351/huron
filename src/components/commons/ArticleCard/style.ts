import { css } from '@emotion/react'

export const header = css({
  marginBottom: '10px'
})

export const articleTitle = css({
  fontSize: '24px',
  marginBottom: '5px'
})

export const list = css({
  margin: '0',
  padding: '0',
  listStyle: 'none',
  display: 'flex',
  gap: '10px'
})

export const section = css({
  marginBottom: '10px'
})

export const text = css({
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  width: '100%',
  display: '-webkit-box',
  WebkitBoxOrient: 'vertical',
  WebkitLineClamp: 3
})
