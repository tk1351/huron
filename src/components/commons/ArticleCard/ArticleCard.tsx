import { FC } from 'react'
import { TextLink } from '../../atoms/TextLink/TextLink'
import { Tag } from '../../atoms/Tag/Tag'
import { articleTitle, header, list, section, text } from './style'
import { Cluster } from '../../atoms/Cluster/Cluster'

type Props = {
  title: string
  createdAt: string
  tags: { id: string; title: string; href: string }[]
  content: string
}

export const ArticleCard: FC<Props> = ({ title, createdAt, tags, content }) => (
  <article>
    <header css={header}>
      <h3 css={articleTitle}>
        <TextLink title={title} href="#" textDecoration="none" />
      </h3>
      <Cluster gap="20px">
        <span>{createdAt}</span>
        <ul css={list}>
          {tags.map((tag) => (
            <li key={tag.id}>
              <Tag title={tag.title} href={tag.href} />
            </li>
          ))}
        </ul>
      </Cluster>
    </header>
    <section css={section}>
      <p css={text}>{content}</p>
    </section>
    <nav>
      <TextLink title="続きを読む" href="#" />
    </nav>
  </article>
)
