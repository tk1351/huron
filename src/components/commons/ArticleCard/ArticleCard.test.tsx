import * as renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import { ArticleCard } from './ArticleCard'

const expectedTitle = 'expectedTitle'
const expectedCreatedAt = 'expectedCreatedAt'
const expectedTags = [
  { id: '1', title: 'react', href: '#' },
  { id: '2', title: 'vue', href: '#' }
]
const expectedContent = 'expectedContent'

const Component = (
  <ArticleCard
    title={expectedTitle}
    createdAt={expectedCreatedAt}
    tags={expectedTags}
    content={expectedContent}
  />
)

describe('Testing ArticleCard Component', () => {
  describe('Render', () => {
    it('正しくArticleCardコンポーネントが表示されていること', () => {
      const App = renderer.create(Component).toJSON()
      expect(App).toMatchSnapshot()
    })
  })

  describe('Props', () => {
    describe('title', () => {
      it('カードのタイトルが正しく渡っていること', () => {
        render(Component)
        expect(screen.getByRole('heading')).toHaveTextContent(expectedTitle)
      })
    })

    describe('createdAt', () => {
      it('作成日が正しく渡っていること', () => {
        render(Component)
        expect(screen.getByText(expectedCreatedAt)).toBeInTheDocument()
      })
    })

    describe('tags', () => {
      it('渡されたオブジェクトの数だけ、tagが正しく表示されていること', () => {
        render(Component)
        const listItems = screen.getAllByRole('listitem')
        expect(listItems).toHaveLength(expectedTags.length)
      })
    })

    describe('content', () => {
      it('本文が正しく渡っていること', () => {
        render(Component)
        expect(screen.getByText(expectedContent)).toBeInTheDocument()
      })
    })
  })
})
