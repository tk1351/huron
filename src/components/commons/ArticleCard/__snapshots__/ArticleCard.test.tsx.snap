// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ArticleCard Component Render 正しくArticleCardコンポーネントが表示されていること 1`] = `
<article>
  <header
    css={
      Object {
        "map": undefined,
        "name": "1r2f04i",
        "next": undefined,
        "styles": "margin-bottom:10px;",
        "toString": [Function],
      }
    }
  >
    <h3
      css={
        Object {
          "map": undefined,
          "name": "1jyzwxm",
          "next": undefined,
          "styles": "font-size:24px;margin-bottom:5px;",
          "toString": [Function],
        }
      }
    >
      <a
        css={
          Object {
            "map": undefined,
            "name": "ytumd6",
            "next": undefined,
            "styles": "text-decoration:none;",
            "toString": [Function],
          }
        }
        href="#"
      >
        expectedTitle
      </a>
    </h3>
    <div
      css={
        Object {
          "map": undefined,
          "name": "k6a5vf",
          "next": undefined,
          "styles": "display:flex;justify-content:flex-start;gap:20px;",
          "toString": [Function],
        }
      }
    >
      <span>
        expectedCreatedAt
      </span>
      <ul
        css={
          Object {
            "map": undefined,
            "name": "1qvxlgp",
            "next": undefined,
            "styles": "margin:0;padding:0;list-style:none;display:flex;gap:10px;",
            "toString": [Function],
          }
        }
      >
        <li>
          <a
            css={
              Object {
                "map": undefined,
                "name": "1dz688v",
                "next": undefined,
                "styles": "background-color:#066163;border-radius:10%;padding:0.2rem;text-decoration:none;",
                "toString": [Function],
              }
            }
            href="#"
          >
            react
          </a>
        </li>
        <li>
          <a
            css={
              Object {
                "map": undefined,
                "name": "1dz688v",
                "next": undefined,
                "styles": "background-color:#066163;border-radius:10%;padding:0.2rem;text-decoration:none;",
                "toString": [Function],
              }
            }
            href="#"
          >
            vue
          </a>
        </li>
      </ul>
    </div>
  </header>
  <section
    css={
      Object {
        "map": undefined,
        "name": "1r2f04i",
        "next": undefined,
        "styles": "margin-bottom:10px;",
        "toString": [Function],
      }
    }
  >
    <p
      css={
        Object {
          "map": undefined,
          "name": "coqqpk",
          "next": undefined,
          "styles": "text-overflow:ellipsis;overflow:hidden;width:100%;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:3;",
          "toString": [Function],
        }
      }
    >
      expectedContent
    </p>
  </section>
  <nav>
    <a
      css={
        Object {
          "map": undefined,
          "name": "1081t4c",
          "next": undefined,
          "styles": "text-decoration:underline;",
          "toString": [Function],
        }
      }
      href="#"
    >
      続きを読む
    </a>
  </nav>
</article>
`;
